### PR TITLE
feat: Update pfe-content-set to accept custom container queries

### PR DESCRIPTION
**Pull request by alazzara**
## Update pfe-content-set to accept custom container queries

Discovery doc:
* https://docs.google.com/document/d/15sfSRWp4OAy2ndbXqHp-nuPjj7kYbyP1zbTqxiHfvo8/edit#heading=h.ughne74p2l4j

Sandbox link:
* https://web-dev-drupal-alazzara.dev.a1.vary.redhat.com/en/pfe-content-set-test

---

### What has changed and why
- Updated pfe-content-set.js - to update offsetWidth based on a value given in a new data attribute called "pfe-breakpoint". This was done to fix and issue with tabs spilling off the page by giving the ability to set when the tabs change to accordions. You can input values with or without px and it will work (the code strips out any non-numeric values). This is to insure the code wont break if someone happens to put 500px instead of just 500.

### Testing instructions
Test cases:
1. If I set the value of the attribute pfe-breakpoint to 500
    a. And I reload the page at window size 300, i expect that
        i. The content set should become accordions
    b. And I reload the page at window size 550, i expect that
        i. The content set should become tabs
2. If I set the value of the attribute pfe-breakpoint to 900
    a. And I reload the page at window size 800, i expect that
        i. The content set should become accordions
    b. And I reload the page at window size 1000, i expect that
        i. The content set should become tabs
3. If I  do not set the value of the attribute pfe-breakpoint 
    a. And I reload the page at window size 600, i expect that
        i. The content set should become accordions
    b. And I reload the page at window size 800, i expect that
        i. The content set should become tabs
4. If I load the page at window size 600, And then I use web inspector to add the attribute pfe-breakpoint 
    a. with a value of 400 after the page has loaded
        i. The content set should become accordions
    b. with a value of 650 after the page has loaded
        i. The content set should become tabs

Edge case:
    1. Test using px and no px in the value of pfe-breakpoint

#### Browser requirements
Your component should work in all of the following environments:

- [x] Latest 2 versions of Edge
- [x] Internet Explorer 11 (should be useable, not pixel perfect)
- [x] Latest 2 versions of Firefox (one on Mac OS, one of Windows OS)
- [x] Firefox 60.7.2 (or latest version for Red Hat Enterprise Linux distribution)
- [x] Latest 2 versions of Chrome (one on Mac OS, one of Windows OS)
- [x] Latest 2 versions of Safari
- [x] Galaxy S9 Firefox
- [x] iPhone X Safari
- [x] iPad Pro Safari
- [x] Pixel 3 Chrome

### Ready-for-merge Checklist

- [x] Expected files: all files in this pull request are related to one feature request or issue (no stragglers)?
- [x] Did browser testing pass?
- [x] Did you update or add any necessary documentation (README.md, WHY.md, etc.)?
- [x] Was this feature demo'd and the design review approved?
- [x] Did you update the CHANGELOG.md file with a summary of this update?

**Be sure to share your updates with the [patternfly-elements-contribute@redhat.com](mailto:patternfly-elements-contribute@redhat.com) mailing list!**

### CODE REVIEW

File: elements/pfe-content-set/src/pfe-content-set.js [hunk 1]
```diff
@@ -24,9 +24,16 @@ class PfeContentSet extends PFElement {
   }
 
   get isTab() {
+    var breakpointValue;
+    if (this.hasAttribute('pfe-breakpoint')) {
+      breakpointValue = this.getAttributeNode('pfe-breakpoint').value;
+      breakpointValue = breakpointValue.replace(/\D/g,'');
+    } else {
+      breakpointValue = 768;
```
Comment (LyndseyR) [inline elements/pfe-content-set/src/pfe-content-set.js]: Should the breakpoint be 767px? I'm use to tablet being 768 and it changes design at tablet. Or below could be `? this.parentNode.offsetWidth >= breakpointValue`?

Comment (starryeyez024) [inline elements/pfe-content-set/src/pfe-content-set.js]:
```suggestion
      breakpointValue = 700;
```

Comment (starryeyez024) [reply elements/pfe-content-set/src/pfe-content-set.js]: 700 as a default approved in PFE touchpoints meeting 👍

File: elements/pfe-content-set/src/pfe-content-set.js [hunk 2]
```diff
@@ -24,9 +24,16 @@ class PfeContentSet extends PFElement {
   }
 
   get isTab() {
+    var breakpointValue;
+    if (this.hasAttribute('pfe-breakpoint')) {
```
Comment (castastrophe) [inline elements/pfe-content-set/src/pfe-content-set.js]: Can you update the single quotes to double for consistency?

File: elements/pfe-content-set/test/pfe-content-set_test.html [hunk 3]
```diff
@@ -177,6 +192,22 @@ <h2 pfe-content-set--header>Heading 1</h2>
 
           assert.equal(alignValue, pfeTabAlignValue);
         });
+
+        test("it should put content into an accordion if the pfe-breakpoint attribute is present and greater than the width of pfe-content-set parent", () => {
```
Comment (castastrophe) [inline elements/pfe-content-set/test/pfe-content-set_test.html]: Great tests!

File: elements/pfe-content-set/demo/index.html [hunk 4]
```diff
@@ -1,328 +1,385 @@
...
-    </style>
-  </head>
-  <body unresolved>
+
+  <link rel="stylesheet" href="../../pfe-styles/dist/pfe-base.css" />
+  <link rel="stylesheet" href="../../pfe-styles/dist/pfe-context.css" />
+  <link rel="stylesheet" href="../../pfe-styles/dist/pfe-layouts.css" />
+
+  <noscript>
+    <link href="../../pfelement/dist/pfelement--noscript.min.css" rel="stylesheet">
```
Comment (castastrophe) [inline elements/pfe-content-set/demo/index.html]: It looks like this is being loaded twice

File: elements/pfe-content-set/demo/index.html [hunk 5]
```diff
@@ -1,328 +1,385 @@
...
-    <link rel="stylesheet" href="https://overpass-30e2.kxcdn.com/overpass.css" />
-    <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1" />
-    <title>PatternFly Element | pfe-content-set Demo</title>
-
-    <!-- Stylesheets for testing light DOM styles.
+
+<head>
+  <meta charset="utf-8">
+  <title>PatternFly Element | pfe-content-set Demo</title>
+  <link rel="stylesheet" href="https://overpass-30e2.kxcdn.com/overpass.css" />
```
Comment (castastrophe) [inline elements/pfe-content-set/demo/index.html]: Do we still need to load overpass?

File: elements/pfe-content-set/demo/index.html [hunk 6]
```diff
@@ -1,328 +1,385 @@
...
+      "../dist/pfe-content-set.umd.js",
+      "../../pfe-cta/dist/pfe-cta.umd.js",
+      "../../pfe-band/dist/pfe-band.umd.js"
+    ])
+
+  </script>
+
+  <noscript>
+    <link href="../../pfelement/dist/pfelement--noscript.min.css" rel="stylesheet">
+    <link href="https://static.redhat.com/libs/redhat/redhat-theme/latest/advanced-theme.css" rel="stylesheet">
```
Comment (castastrophe) [inline elements/pfe-content-set/demo/index.html]: I think the theme would be loaded outside the noscript tag

File: elements/pfe-content-set/demo/index.html [hunk 7]
```diff
@@ -1,328 +1,385 @@
...
+      </pfe-content-set>
+    </div>
+  </section>
+
+  <script>
+    var btn = document.querySelector("#addHeaderAndPanelBtn");
+    var pfeContentSet = document.querySelector("#dynamic");
 
-    <button id="addHeaderAndPanelBtn">Add Header &amp; Panel</button>
+    btn.addEventListener("click", function () {
```
Comment (castastrophe) [inline elements/pfe-content-set/demo/index.html]: Cool functionality!!

File: elements/pfe-content-set/src/pfe-content-set.js (hunk 2 above)
Comment (mwcz) [reply elements/pfe-content-set/src/pfe-content-set.js]: Related: prettier should be running on commit, but it doesn't seem to be.  I'll look into it.

Comment (alazzara) [reply elements/pfe-content-set/src/pfe-content-set.js]: updated.

File: elements/pfe-content-set/demo/index.html (hunk 5 above)
Comment (starryeyez024) [reply elements/pfe-content-set/demo/index.html]: Need to? Probably not. Though it looks nicer.. any reason why we shouldnt?

Comment (castastrophe) [reply elements/pfe-content-set/demo/index.html]: Oh I thought we had switched the font away from overpass.

Comment (starryeyez024) [reply elements/pfe-content-set/demo/index.html]: ah, fair. switched to redhatfont

File: elements/pfe-content-set/demo/index.html (hunk 6 above)
Comment (starryeyez024) [reply elements/pfe-content-set/demo/index.html]: thx! fixed

File: elements/pfe-content-set/demo/index.html (hunk 4 above)
Comment (starryeyez024) [reply elements/pfe-content-set/demo/index.html]: thanks, fixed!

File: CHANGELOG-prerelease.md [hunk 8]
```diff
@@ -2,9 +2,10 @@
 
 Tag: [v1.0.0-prerelease.37](https://github.com/patternfly/patternfly-elements/releases/tag/v1.0.0-prerelease.37)
 
+- [](https://github.com/patternfly/patternfly-elements/commit/) Update pfe-content-set to accept custom container queries
 - [6a41811](https://github.com/patternfly/patternfly-elements/commit/6a418112668ba580918ac4a8b4b54e8df05c1155) fix: reference error when slot is missing in pfe-navigation
 - [ff859a5](https://github.com/patternfly/patternfly-elements/commit/ff859a5c2b62ae24225f0031f69b5bc050c59470) fix: accordion accessibility improvements; aria-roles corrected
-- [](https://github.com/patternfly/patternfly-elements/commit/) fix: pfe-nav & pfe-cta hotfixes
+- [](https://github.com/patternfly/patternfly-elements/commit/) fix: pfe-nav & pfe-cta hotfixes aria-roles corrected
```
Comment (starryeyez024) [inline CHANGELOG-prerelease.md]: @alazzara this line should be:

```suggestion
- [](https://github.com/patternfly/patternfly-elements/commit/) fix: pfe-nav & pfe-cta hotfixes
```

**Comment by starryeyez024** [comment]
@castastrophe this PR is stuck on your requested changes, can you approve when you have a moment? Thx!

File: elements/pfe-content-set/src/pfe-content-set.js (hunk 2 above)
Comment (mwcz) [reply elements/pfe-content-set/src/pfe-content-set.js]: Related: Attempting to fix prettier in #726.

File: examples/index.html [hunk 9]
```diff
@@ -59,6 +59,7 @@ <h1>Demo pages</h1>
 				<pfe-cta pfe-priority="secondary" pfe-variant="wind"><a href="../elements/pfe-health-index/demo">pfe-health-index</a></pfe-cta>
 				<pfe-cta pfe-priority="secondary" pfe-variant="wind"><a href="../elements/pfe-icon/demo">pfe-icon</a></pfe-cta>
 				<pfe-cta pfe-priority="secondary" pfe-variant="wind"><a href="../elements/pfe-icon-panel/demo">pfe-icon-panel</a></pfe-cta>
+				<pfe-cta pfe-priority="secondary" pfe-variant="wind"><a href="../elements/pfe-layouts/demo">pfe-layouts</a></pfe-cta>
```
[the remark this answers is not in the document]
Comment (castastrophe) [reply examples/index.html]: @starryeyez024 @alazzara bump ^

[the remark this answers is not in the document]
Comment (starryeyez024) [reply examples/index.html]: fixed!
